src/Parser/FormUrlencoded.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace React\Http\Parser;
+
+use Evenement\EventEmitterTrait;
+use React\Http\Request;
+
+class FormUrlencoded implements ParserInterface
+{
+    use EventEmitterTrait;
+
+    /**
+     * @var string
+     */
+    protected $buffer = '';
+
+    /**
+     * @var Request
+     */
+    protected $request;
+
+    /**
+     * @var bool|integer
+     */
+    protected $contentLength = false;
+
+    /**
+     * @param Request $request
+     */
+    public function __construct(Request $request)
+    {
+        $this->request = $request;
+
+        $headers = $this->request->getHeaders();
+        $headers = array_change_key_case($headers, CASE_LOWER);
+
+        $this->contentLength = $headers['content-length'];
```

Comment:
And if not existing throw an exception I assume. That check is already done in the form parser factory.